src/main/java/org/folio/mr/domain/entity/MediatedRequestEntity.java

```diff
@@ -23,11 +23,19 @@
 import jakarta.persistence.Id;
 import jakarta.persistence.OneToMany;
 import jakarta.persistence.Table;
-import lombok.Data;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.With;
 
-@Data
 @Table(name = "mediated_request")
 @Entity
+@Setter
+@Getter
+@With
+@AllArgsConstructor
+@NoArgsConstructor
```

Comment:
Perhaps, you forgot to put equalsAndHashcode annotation, since you replace Data annotation